zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +136,160 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() {
+    // Test multiple small transactions to fill a block max size
+    // A dummy transaction
+    let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
+
+    // A block header
+    //let blockheader = generate_block_header();
+    let blockheader = zebra_test::generate::block_header().clone();
+
+    // Serialize header
+    let mut data_header = Vec::new();
+    blockheader
+        .zcash_serialize(&mut data_header)
+        .expect("Block header should serialize");
+
+    // Calculate the number of transactions we need
+    let mut max_transactions_in_block =
+        (MAX_BLOCK_BYTES as usize - data_header.len()) / *&zebra_test::vectors::DUMMY_TX1[..].len();
+
+    // Create transactions to be just below the limit
+    let mut many_transactions = std::iter::repeat(Arc::new(tx.clone()))
+        .take(max_transactions_in_block)
+        .collect::<Vec<_>>();
+
+    // Add the transactions into a block
+    let mut block = Block {
+        header: blockheader,
+        transactions: many_transactions.clone(),
+    };
```

Comment:
Suggestion: Can you move large block generation to its own function in a `zebra_chain::test::generate` module?

I'd like to be able to create large blocks in other tests.

The functions would look something like:
```
fn large_multi_transaction_block() -> Block
fn large_single_transaction_block() -> Block
fn oversized_multi_transaction_block() -> Block
fn oversized_single_transaction_block() -> Block
```

There might be some way to share code between these functions, but let's do that in a future PR.